panther_lights/src/animation/battery_animation.py

```diff
@@ -6,16 +6,17 @@
 
 
 class BatteryAnimation(Animation):
-
     ANIMATION_NAME = 'battery_animation'
 
     def __init__(self, *args, **kwargs) -> None:
         super().__init__(*args, **kwargs)
 
         self._anim = np.zeros((self._anim_len, self.num_led))
-        self._h_min = 0.0
-        self._h_max = 120.0
+        self._h_min = 0.0  # red color
+        self._h_max = 0.4  # green color
         self._resolution = 100
+        self._start_fade = 0.80 * self._resolution
+        self._end_anim = 0.90 * self._resolution
```

Comment:
Starting fade before `_end_anim` won't work for `battery_percent = 1.0`. Animation won't have enough time to finish. I think starting fade at `_end_anim` point should be enough to provide smooth fade.
Also, round to one decimal place should be enough.

Reply:
I leave dark panel for last 5% of time, prevents sudden jump